components/form/builder/src/index.js

```diff
@@ -69,6 +66,10 @@ const FormBuilder = ({
 
     clearTimeout(timerShowSpinner)
     setStateFields(nextStateFields)
+    onChange({
+      ...fieldsToObject(nextFields),
```

Comment:
The line that was removed, was this one: https://github.com/SUI-Components/adevinta-spain-components/pull/225/files#diff-b44d4b0d39148c1e3df23e43a1c54244L73 Pay attention that the fields that are passed were: `nextStateFields` instead of `nextFields`.

Reply:
If passing `nextFields` instead of `nextStateFields` you are sending the 'old' params in the onChange callback instead of the new ones.

Reply:
uaau sorry, FIXED!

Reply:
In fact, the duplication of onChange call was made intentionally because the function `reducerWithRules` can take some time depending on the number of rules. So the component that uses the form builder can obtain some quick feedback first and then apply all the rules and send onChange feedback again. I approved last PR (removing duplicates) without remembering this intentional behavior but it seems that anybody has been affected.

Form builder is hard and with 0 tests, but we should consider (me first) start writting tests in this component so we have the security that we are not introducing breaking changes since it is used in production in Milanuncios or creating new majors when we are not sure that the changes can break exisiting implementations so we can test new PR's. Otherwise it will not be possible to introduce new changes in the future without having the security that we are not breaking things. What do you think?